text/0000-scoped-custom-elements.md

```diff
@@ -0,0 +1,109 @@
+
+`@scope/custom-element/src/CustomElement.js`
+
+```javascript
+export declare class CustomElement extends HTMLElement {
+  constructor() {
+    const registry = CustomElement.registry;
+    this.attachShadow({mode: 'open', registry});
+  }
+}
+```
+
+The LWC component imports the custom element, passes the handler to the custom element, and registers it.
+
+```javascript
+import { LightningElement } from 'lwc';
+import { CustomElement } from '@scope/custom-element';
+import { registry } from './registry.js';
+CustomElement.registry = registry;
+registry.define('custom-element', CustomElement);
+
+export default class Example extends LightningElement {
+  ...
+}
+```
+
+```html
+<template>
+  <custom-element></custom-element>
+</template>
+```
+
+The whole registration process of external custom elements could be done in the `registry.js` file instead of inside an LWC component to avoid name duplicates.
+
+Finally, the LWC component uses the same registry to register itself in the same registry as the rest of the components.
+
+### Polyfills
+
+There is [Scoped CustomElementRegistry polyfill](https://github.com/webcomponents/polyfills/tree/master/packages/scoped-custom-element-registry) that can be used as a base for the implementation. However, it is a work in progress.
+
+## Drawbacks
+
+Why should we *not* do this? Please consider:
+
+- implementation cost, both in term of code size and complexity
+- whether the proposed feature can be implemented in user space
+- the impact on teaching people the proposed feature
+- integration of this feature with other existing and planned features
+- cost of migrating existing applications (is it a breaking change?)
+
+I see a problem with the static field with the registry on the element. Generally I hate an idea of having static fields to pass references to a crucial part of the registration process. Even though it can't be altered by another application, an author can easily override it with another registry. This could be potentially beneficial in cases when an LWC component is made to be sharable so this component's registry is also dynamic. But it could potentially mean that the scope of the registration change without author realizing what is happening.
+
+## Alternatives
+
+Moving existing applications from CE to LWC. However, the cost of doing so is huge. We also can't expect all customers to migrate their own stack to LWC. Furthermore, LWC is not particularly friendly towards being used as a component in non-LWC projects so in many cases migration to LWC is impossible.
+
+## Adoption strategy
+
+If we implement this proposal, how will existing Lightning Web Stack developers adopt it? Is
+this a breaking change? Can we write a codemod? Should we coordinate with
+other projects or libraries?
+
+TBD
+
+# How we teach this
+
+This is set to become a web platform standard. When it become a standard then there's no need for training materials other than code examples.
```

Comment:
there is no such as as "set to become a web platform standard", never count on that, wait for it to land in 2 browsers, then you are mostly safe.